### PR TITLE
Added action to set number of products in storage

### PR DESCRIPTION
**Pull request by TomaszQr**

### CODE REVIEW

**Comment by mariuszkrzaczkowski** [comment]
## Migration Recommended

It appears you are still using the CoPilot OAuth App - we recommend you [migrate to the GitHub App](https://github.com/apps/blackduck-copilot/installations/new/permissions?suggested_target_id=10201412&repository_ids[]=28060246)

**Comment by mariuszkrzaczkowski** [comment]
## Black Duck Security Report

Merging #11246 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/11246)

File: modules/IStorages/models/Record.php [hunk 1]
```diff
@@ -38,9 +38,33 @@ public function getHierarchy()
...
+
+	/**
+	 * Function updates number of product in storage.
+	 *
+	 * @param int   $relatedRecordId - Product Id
+	 * @param float $qty
+	 */
+	public function updateQtyProducts(int $relatedRecordId, float $qty): bool
+	{
+		$isExists = (new \App\Db\Query())->from('u_#__istorages_products')->where(['crmid' => $this->getId(), 'relcrmid' => $relatedRecordId])->exists();
```
Comment (rskrzypczak) [inline modules/IStorages/models/Record.php]: ?

File: modules/IStorages/models/Record.php [hunk 2]
```diff
@@ -38,9 +38,33 @@ public function getHierarchy()
...
 		}
 		return parent::getDisplayValue($fieldName, $record, $rawText, $length);
 	}
+
+	/**
+	 * Function updates number of product in storage.
+	 *
+	 * @param int   $relatedRecordId - Product Id
+	 * @param float $qty
+	 */
```
Comment (rskrzypczak) [inline modules/IStorages/models/Record.php]: @return bool

File: modules/IStorages/actions/RelationAjax.php [hunk 3]
```diff
@@ -13,21 +13,42 @@ public function __construct()
...
+		$sourceModule = $request->getModule();
+		if (!\App\Privilege::isPermitted($sourceModule, 'SetQtyProducts')) {
+			throw new \App\Exceptions\NoPermittedToRecord('ERR_NO_PERMISSIONS_TO_ACTION', 406);
+		}
+		$recordModel = Vtiger_Record_Model::getInstanceById($request->getInteger('src_record'), $sourceModule);
+		if (!$recordModel->isViewable()) {
+			throw new \App\Exceptions\NoPermittedToRecord('ERR_NO_PERMISSIONS_FOR_THE_RECORD', 406);
+		}
+		$response = new Vtiger_Response();
+		$response->setResult((bool) $recordModel->updateQtyProducts($request->getInteger('record'), $request->getByType('qty', 'NumberInUserFormat')));
```
Comment (rskrzypczak) [inline modules/IStorages/actions/RelationAjax.php]:
```suggestion
		$response->setResult($recordModel->updateQtyProducts($request->getInteger('record'), $request->getByType('qty', 'NumberInUserFormat')));
```

File: modules/IStorages/models/Record.php [hunk 4]
```diff
@@ -38,9 +38,33 @@ public function getHierarchy()
...
+				->execute();
+		} else {
+			$status = App\Db::getInstance()->createCommand()
+				->insert('u_#__istorages_products', [
+					'crmid' => $this->getId(),
+					'relcrmid' => $relatedRecordId,
+					'qtyinstock' => $qty,
+				])->execute();
+		}
+		return $status;
```
Comment (rskrzypczak) [inline modules/IStorages/models/Record.php]:
```suggestion
		return (bool) $status;
```

File: public_html/layouts/basic/modules/IStorages/resources/RelatedList.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/* {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} */
+'use strict';
+
+Vtiger_RelatedList_Js(
+	'IStorages_RelatedList_Js',
+	{
+			registerEditQty: function () {
+			let thisInstance = this;
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/IStorages/resources/RelatedList.js]: code formatting

**Pull request merged**